definitions/npm/ramda_v0.x.x/flow_v0.62.x-/ramda_v0.x.x.js

```diff
@@ -407,13 +407,25 @@ declare module ramda {
     ) => UnaryFn<A, C>) &
     (<A, B>(ab: UnaryFn<A, B>) => UnaryFn<A, B>);
 
-  declare type Filter = (<K, V, T: Array<V> | { [key: K]: V }>(
-    fn: UnaryPredicateFn<V>,
-    xs: T
-  ) => T) &
-    (<K, V, T: Array<V> | { [key: K]: V }>(
-      fn: UnaryPredicateFn<V>
-    ) => (xs: T) => T);
+  // This kind of filter allows us to do type refinement on the result, but we
+  // still need Filter so that non-refining predicates still pass a type check.
+  declare type RefineFilter =
+    & (<K, V, P: $Pred<1>, T: Array<V> | { [key: K]: V }>(
```

Comment:
Nicely spotted on the `$Pred` and `$Refine`, didn't know those existed.

Reply:
Thanks! It's still officially undocumented per https://github.com/facebook/flow/issues/2464 - but [this comment](
https://github.com/facebook/flow/issues/2464#issuecomment-251995029) links to the comment block where there's kind of enough to go off of. To be honest, it took me quite a long time to understand how it was intended to be applied. But hey, we have a living example now!